src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStrip.cs

```diff
@@ -1970,27 +1942,31 @@ protected override void Dispose(bool disposing)
                 SuspendLayout();
                 overflow?.SuspendLayout();
 
-                // if there's a problem in config, don't be a leaker.
+                // If there's a problem in config, don't be a leaker.
                 SetToolStripState(STATE_DISPOSINGITEMS, true);
                 _lastMouseDownedItem = null;
 
-                HookStaticEvents(/*hook=*/false);
+                HookStaticEvents(hook: false);
 
-                if (Properties.GetObject(s_propToolStripPanelCell) is ToolStripPanelCell toolStripPanelCell)
+                if (Properties.TryGetValue(s_propToolStripPanelCell, out ToolStripPanelCell? toolStripPanelCell))
                 {
                     toolStripPanelCell.Dispose();
+                    Properties.RemoveValue(s_propToolStripPanelCell);
                 }
 
                 _cachedItemHdcInfo?.Dispose();
 
                 _mouseHoverTimer?.Dispose();
 
-                ToolTip? toolTip = (ToolTip?)Properties.GetObject(s_propToolTip);
-                toolTip?.Dispose();
+                if (Properties.TryGetValue(s_propToolTip, out ToolTip? toolTip))
+                {
+                    toolTip?.Dispose();
```

Comment:
Do we still need ? here?
```suggestion
                    toolTip.Dispose();
```